packages/react/src/lib/createComponent.tsx

```diff
@@ -18,7 +18,7 @@ export interface CreateComponentConfig<P> {
   render: (config: RenderResultConfig<P>, props: P) => React.ReactNode
 }
 
-export type CreateComponentReturnType<P> = React.SFC<P> & {
+export type CreateComponentReturnType<P> = React.FunctionComponent<P> & {
```

Comment:
```suggestion
export type CreateComponentReturnType<P> = React.FC<P> & {
```
nit: can be just `FC`

Reply:
might change for now (to be consistent with the rest of the code), but, frankly, not completely sure that we should use short alias name version for some types (`FC` instead of `FunctionComponent`) and, at the same time, would have to use long names for the others. Would rather expect that we would use explicit `FunctionComponent` everywhere - what do you think?

Reply:
`FC` has one big advantage, it's more shorter :)

```tsx
const LayoutExample: React.FunctionalComponent<LayoutExampleProps> = ({ vertical, horizontal })
// vs
const LayoutExample: React.FC<LayoutExampleProps> = ({ vertical, horizontal })
```

Reply:
this has a disadvantage - it is not that explicit :) And the key differentiator to me is the fact that for the rest of React types we still would need to use log versions of type names. Thus, I would rather stay with consistency and expressivity

Reply:
just as an example to compare

### now
```js
createShorthandFactory<TFunctionComponent extends React.FunctionComponent>(
  Component: TFunctionComponent,
  mappedProp?: keyof PropsOf<TFunctionComponent>,
)
```

### changed
```js
createShorthandFactory<TFC extends React.FC>(
  Component: TFC,
  mappedProp?: keyof PropsOf<TFC>,
)
```

Reply:
I vote for descriptive (longer) name